### PR TITLE
Add form-field state

### PR DESCRIPTION
**Pull request by Pante**
**Describe the changes**

**Checklist**
- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md).
- [ ] I have included the relevant unit/golden tests.
- [ ] I have included the relevant samples.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the [CHANGELOG.md](../forui/CHANGELOG.md) if necessary.

### CODE REVIEW

File: forui/lib/src/widgets/select_group/select_group.dart [hunk 1]
```diff
@@ -38,43 +40,53 @@ class FSelectGroup<T> extends StatelessWidget {
...
+  FSelectGroup({
     required this.controller,
     required this.items,
     this.style,
     this.label,
     this.description,
     this.error,
+    super.onSaved,
+    super.validator,
+    super.initialValue,
```
Comment (kawaijoe) [inline forui/lib/src/widgets/select_group/select_group.dart:52]: How does this work with a controller?

**Review by kawaijoe — CHANGES_REQUESTED** [review]
Also on a side note, the build appears to be failing.

File: forui/lib/src/widgets/select_group/select_group.dart (hunk 1 above)
Comment (Pante) [reply forui/lib/src/widgets/select_group/select_group.dart:52]: * `onSaved` is called when the values are saved, no direct interaction with the controller.
* `validator` is called when the fields are validated, no direct interaction with the controller.
* `initialValue` is used when the `reset()` method is called and when the state's internal value is set to null.

In most cases, the state and controller are synced via `_handleControllerChanged` and `didChange(...)` methods.

File: forui/lib/src/foundation/form_field.dart [hunk 2]
```diff
@@ -72,20 +72,12 @@ abstract class FFormField<T> extends StatelessWidget {
...
   Widget build(BuildContext context) => FormField<T>(
         onSaved: onSave,
-        // TODO: Directly use forceErrorText when available. https://api.flutter.dev/flutter/widgets/FormField/forceErrorText.html
-        validator: forceErrorText == null ? validator : (_) => forceErrorText,
+        validator: validator,
         initialValue: initialValue,
         enabled: enabled,
-        autovalidateMode:
-            forceErrorText == null ? autovalidateMode : AutovalidateMode.always, // Workaround for forceErrorText.
+        autovalidateMode: autovalidateMode, // Workaround for forceErrorText.
```
Comment (kawaijoe) [inline forui/lib/src/foundation/form_field.dart:78]: I think we can remove the comment

File: forui/lib/src/widgets/select_group/select_group_controller.dart [hunk 3]
```diff
@@ -17,6 +17,13 @@ abstract class FSelectGroupController<T> with ChangeNotifier {
   /// The currently selected values.
   Set<T> get values => {..._values};
 
+  @protected
+  set values(Set<T> values) => {
```
Comment (kawaijoe) [inline forui/lib/src/widgets/select_group/select_group_controller.dart:21]: I might have missed it, but is this used anywhere? All good if it's used.

**Review by kawaijoe — APPROVED** [review]

**Pull request merged**